### DISCUSSION TITLE
integração com compose kotlin e java15

### DISCUSSION THREAD

**Discussion by zuza07** [wmixvideo/nfe#801] [Geral]
Olá pessoal tudo bem ? Fiz uma pequeno programa de integração que lê a XML e envia para a SEFAZ, porém em tempo de DEBUG funciona normal pelo menos a captura do status, ao testar o programa para distribuir apresenta o seguinte erro: 
./gradlew runDistributable

Exception in thread "AWT-EventQueue-0" java.lang.NoClassDefFoundError: java/rmi/RemoteException
        at com.fincatto.documentofiscal.nfe400.webservices.WSFacade.<init>(WSFacade.java:58)
        at funcoes.ffLote.verStatus(ffLote.kt:50)
        at form.HomeScreen.clickStatusNFe(HomeScreen.kt:449)
        at form.HomeScreen.access$clickStatusNFe(HomeScreen.kt:50)
        at form.HomeScreen$show$3$2$1.invoke(HomeScreen.kt:138)
        at form.HomeScreen$show$3$2$1.invoke(HomeScreen.kt:135)

O programa apresenta a seguinte mensagem java/rmi/RemoteException, já tentei de tudo com o fonte no local ou mesmo baixando com gradle 2 dias louco em cima disso e não compreendo, minha versão anterior com java 1.8 e FMX funciona, já nessa apenas em DEBUG.

Se surgir o interesse de alguém ajudar, pode comentar inclusive posso enviar os fontes é um programa pequeno (deixa o e-mail), suave para compilar.
Desde já grato!

**Comment by fincatto** [comment]
Qual a versão do java você está usando atualmente?

**Reply by zuza07** [reply]
Estou usando oracle jdk version 15.0.2

**Reply by fincatto** [reply]
Consegue fazer um teste com a jdk11?

**Reply by zuza07** [reply]
Fiz a instalação, compilou funcionou, ./gradlew run também funcionou, porém o ./gradlew createDistributable falhou.

[zuza@fedora-z integra01]$ ./gradlew createDistributable
> Task :checkRuntime FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':checkRuntime'.
> Packaging native distributions requires JDK runtime version >= 15
  Actual version: '11'
  Java home: /usr/java/jdk-11.0.13

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Amanhã vou pesquisar se dá para rodar esse aplicativo sem necessidade do JAVA15 e sem rodar o comando 
./gradlew createDistributable, essa exigência deve estar atrelada ao compose, eu já tinha lido em algum lugar que teria que usar no mínimo a versão Java15.

Mas blz, valew, bom descanso !

**Comment by zuza07** [comment]
bom pessoal, volto aqui somente para informar que não foi possível gerar um arquivo de distribuição do App feito em kotlin + compose (feito a fim de estudo testes para ambiente de produção) com o componente, penso que deva ser devido a necessidade de rodar usando gradlew + Java15, enfim tenho a minha outra App rodando vou mante-la da forma como está ... e vou ficar de olho para um futuro upgrade do componente para o Java 15.
grato pelos esclarecimentos e pela solução, até + abraços.
